apps/elixir_ls_debugger/lib/debugger/server.ex

```diff
@@ -887,23 +867,26 @@ defmodule ElixirLS.Debugger.Server do
     end)
   end
 
-  defp ensure_thread_id(state = %__MODULE__{}, pid) when is_pid(pid) do
+  defp ensure_thread_id(state = %__MODULE__{}, pid, new_ids) when is_pid(pid) do
```

Comment:
maybe keeping the current thread id in the process dictionary and guarding that with functions might be easier than placing it in the state?

Reply:
```elixir
def current_thread_id do 
  Process.get(:current_thread_id, 0)
end 

def next_thread_id do 
   next_id = current_thread_id() + 1
   Process.put(:current_thread_id, next_id)
   next_id
end
```

Reply:
I wouldn't do it as part of this PR
Created https://github.com/elixir-lsp/elixir-ls/issues/843